src/components/avatar-rectangle/avatar-rectangle.tsx

```diff
@@ -0,0 +1,49 @@
+import { useState, type FC, type HTMLAttributes } from 'react'
+import {
+  ElAvatarRectangleWrapper,
+  ElAvatarRectCommercialPlaceholder,
+  ElAvatarRectangleResidentialPlaceholder,
+  ElAvatarRectBottomPlaceholder,
+  ElAvatarRectBottomPlaceholderSmall,
+} from './styles'
+
+export interface AvatarRectangle extends HTMLAttributes<HTMLSpanElement> {
+  variant: 'residential' | 'commercial'
+  size: 'medium' | 'small'
+  src?: string
+  alt?: string
+}
+
+export const AvatarRectangle: FC<AvatarRectangle> = ({
+  size = 'medium',
+  variant = 'residential',
+  src,
+  alt,
+  ...props
+}) => {
+  const [isError, setIsError] = useState(false)
+
+  const handleError = () => {
+    setIsError(true)
+  }
+
+  const imageValid = !!src && !isError
+
+  return (
```

Comment:
Ah, I thought the fallback would be the placeholder version. In that case, if the `fallbackUri` also results in an error, should I show the placeholder or just use the alt text? 🤔

Reply:
Yes, the "fallback" is the "placeholder". With approach (1), the fallback URI would never fail because we'd use a data URI for the image, but that's beside the point. My comment above is not suggestion we should use approach (1), I'm just trying to show the importance of making our code easily _readable_. In both snippets, the intention of the code is very clear, whereas the code below is not.

Reply:
ok, I so the fallback will be the defined svg, I doubt that I can download the full svg of the residential version (last time I tried I only got the icon not full) but I will ask the Design team first, once I got them that would be great 👍